src/pymatgen/core/structure.py

```diff
@@ -48,6 +48,7 @@
     from collections.abc import Callable, Iterable, Iterator, Sequence
     from typing import Any, ClassVar, SupportsIndex, TypeAlias
 
+    import moyopy
```

Comment:
This makes moyopy an actual dependency. Not an optional one. This needs to be in a try catch.

Reply:
it's inside `if TYPE_CHECKING:`, so only imported at type-checking time